avocado/utils/filelock.py

```diff
@@ -0,0 +1,110 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2016
+# Author: Amador Pahim <apahim@redhat.com>
+
+
+import errno
+import os
+import time
+
+
+class AlreadyLocked(Exception):
+    pass
+
+
+class LockFailed(Exception):
+    pass
+
+
+class LockFile(object):
+    """
+    Creates a lock for a file.
+
+    If the lock file already exists and it contains the current process
+    pid in it, we wait until the timeout for the lock to be released,
+    raising an AlreadyLocked exception when the timeout is reached.
+
+    If the lock file already exists and it contains a pid of another
+    running process, we raise an NotMyLock exception.
```

Comment:
I don't see a `NotMyLock` exception anywhere.  Looks like it's become `AlreadyLocked`, right?